src/silo/common/string_utils.test.cpp

```diff
@@ -0,0 +1,37 @@
+#include "silo/common/string_utils.h"
+
+#include <gtest/gtest.h>
+
+using silo::splitBy;
+
+TEST(splitBy, correctSplit) {
+   const std::string input("ABC,DEF,ADS,");
+   const std::string delimiter(",");
+
+   const auto result = splitBy(input, delimiter);
+   EXPECT_EQ(result, std::vector<std::string>({"ABC", "DEF", "ADS", ""}));
+}
+
+TEST(splitBy, correctSplit2) {
+   const std::string input("ABC");
+   const std::string delimiter(",");
+
+   const auto result = splitBy(input, delimiter);
+   EXPECT_EQ(result, std::vector<std::string>({"ABC"}));
+}
+
+TEST(splitBy, correctSplit3) {
+   const std::string input("ABC.*");
+   const std::string delimiter(".*");
+
+   const auto result = splitBy(input, delimiter);
+   EXPECT_EQ(result, std::vector<std::string>({"ABC", ""}));
+}
+
+TEST(splitBy, correctWithEmptyString) {
+   const std::string input;
+   const std::string_view delimiter(",");
+
+   const auto result = splitBy(input, delimiter);
+   EXPECT_EQ(result, std::vector<std::string>{""});
```

Comment:
Isn't this wrong? I would expect to see `["", ""]`.

This is what Javascript does:
```
        expect(','.split(',')).toEqual(['', '']);
        expect('a,'.split(',')).toEqual(['a', '']);
        expect(',b'.split(',')).toEqual(['', 'b']);
        expect('a,b'.split(',')).toEqual(['a', 'b']);
        expect('a,b,'.split(',')).toEqual(['a', 'b', '']);
```

Reply:
`input` contains an empty string `""`, so splitting by `,` (`delimiter`) should return `[""]` (In JS: `expect("".split(",").toEqual(""));`

Reply:
Ah sorry, I misread. I still find C++ syntax confusing sometimes.